File: dev-docs/snippets/document-blob/upload-file.ts
import type { JsBaoClient } from "js-bao-wss-client";

export async function uploadFile(
  client: JsBaoClient,
  documentId: string,
  data: Uint8Array,
) {
  // #region example
  const blobs = client.document(documentId).blobs();
  // Like upload(), but queues for background transfer if the upload queue is active.
  const { blobId, numBytes } = await blobs.uploadFile(data, {
    filename: "large.bin",
    contentType: "application/octet-stream",
  });
  // #endregion example
  return { blobId, numBytes };
}

File: dev-docs/snippets/document-blob/upload-file.swift
import Foundation
import JsBaoClient

func uploadFile(
  client: JsBaoClient,
  documentId: String,
  data: Data
) async throws {
  // #region example
  let blobs = client.documents.blobs(documentId: documentId)
  // Like upload(), but returns the narrowed queue shape ({ blobId, numBytes,
  // bytesTransferred }) and queues for background transfer when the immediate
  // upload can't complete.
  let result = try await blobs.uploadFile(
    data: data,
    options: BlobUploadSourceOptions(
      filename: "large.bin",
      contentType: "application/octet-stream"
    )
  )
  let blobId = result.blobId
  let numBytes = result.numBytes
  // #endregion example
  _ = (blobId, numBytes)
}
